screeps-game-api/src/objects/impls/room.rs

```diff
@@ -168,3 +273,158 @@ impl PartialEq for Room {
+            max_rooms: 16,
+            range: 0,
+            plain_cost: 1,
+            swamp_cost: 5,
+        }
+    }
+}
+
+impl FindOptions<'static, fn(String, CostMatrix) -> Option<CostMatrix<'static>>> {
+    /// Creates default SearchOptions
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl<'a, F> FindOptions<'a, F>
+where
+    F: Fn(String, CostMatrix) -> Option<CostMatrix<'a>>,
+{
+    /// Sets whether the algorithm considers creeps as walkable. Default: False.
+    pub fn ignore_creeps(mut self, ignore: bool) -> Self {
+        self.ignore_creeps = ignore;
+        self
+    }
+
+    /// Sets whether the algorithm considers destructible structure as
+    /// walkable. Default: False.
+    pub fn ignore_destructible_structures(mut self, ignore: bool) -> Self {
+        self.ignore_destructible_structures = ignore;
+        self
+    }
+
+    /// Sets cost callback - default `|_, _| {}`.
+    pub fn cost_callback<'b, F2>(self, cost_callback: F2) -> FindOptions<'b, F2>
+    where
+        F2: Fn(String, CostMatrix) -> Option<CostMatrix<'b>>,
+    {
+        let FindOptions {
+            ignore_creeps,
+            ignore_destructible_structures,
+            cost_callback: _,
+            max_ops,
+            heuristic_weight,
+            serialize,
+            max_rooms,
+            range,
+            plain_cost,
+            swamp_cost,
+        } = self;
+        FindOptions {
+            ignore_creeps,
+            ignore_destructible_structures,
+            cost_callback,
+            max_ops,
+            heuristic_weight,
+            serialize,
+            max_rooms,
+            range,
+            plain_cost,
+            swamp_cost,
+        }
+    }
+
+    /// Sets maximum ops - default `2000`.
+    pub fn max_ops(mut self, ops: u32) -> Self {
+        self.max_ops = ops;
+        self
+    }
+
+    /// Sets heuristic weight - default `1.2`.
+    pub fn heuristic_weight(mut self, weight: f64) -> Self {
+        self.heuristic_weight = weight;
+        self
+    }
+
+    /// Sets whether the returned path should be passed to `Room.serializePath`.
+    pub fn serialize(mut self, s: bool) -> Self {
+        self.serialize = s;
+        self
+    }
+
+    /// Sets maximum rooms - default `16`, max `16`.
+    pub fn max_rooms(mut self, rooms: u32) -> Self {
+        self.max_rooms = rooms;
+        self
+    }
+
+    pub fn range(mut self, k: u32) -> Self {
+        self.range = k;
+        self
+    }
+
+    /// Sets plain cost - default `1`.
+    pub fn plain_cost(mut self, cost: u8) -> Self {
+        self.plain_cost = cost;
+        self
+    }
+
+    /// Sets swamp cost - default `5`.
+    pub fn swamp_cost(mut self, cost: u8) -> Self {
+        self.swamp_cost = cost;
+        self
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Step {
+    x: u32,
+    y: u32,
+    dx: i32,
+    dy: i32,
+    direction: Direction
+}
+
+js_deserializable!{Step}
+
+#[derive(Debug, Deserialize)]
```

Comment:
The `Deserialize` implementation for this will probably need a `#[serde(untagged)]` annotation since the names `VectorizedPath` and `SerializedPath` aren't part of the API return.

Reply:
Good catch